### DISCUSSION TITLE
Contribute vscode-trace-extension repository to the Eclipse CDT-cloud project

### DISCUSSION THREAD

**Discussion by bhufmann** [eclipse-cdt-cloud/cdt-cloud#26] [General]
Hello,

The Trace Compass team would like to contribute the `vscode-trace-extension` [1] to the `Eclipse CDT.cloud` project  so that the further development can be done under the Eclipse open source foundation. The `Eclipse CDT.cloud` is already the home of the main components that are used by the `vscode-trace-extension`.

The goal of this project is to provide the trace-viewer capabilities as VS Code plugin so that all VS Code compatible tools like Theia, VSCodium, VS Code etc. can install and take advantage of the trace viewer functionality, and not "just" Theia based products. There has been also an increasing interest of this VS Code plug-in in the community.

The `vscode-trace-extension` repository only contains code to integrate with VS Code APIs. All core, trace-viewing source code are provided by the Theia-independent components of `theia-trace-extension` [2] which are deployed to NPM [3] and [4].

Issue tracker [5] was created to inform the contributors to 'vscode-trace-extension' about this intention.

I'd like to start the contribution process in the coming week(s). Please let me know if you have any questions and concerns.

Best Regards
Bernd

[1] https://github.com/theia-ide/vscode-trace-extension
[2] https://github.com/eclipse-cdt-cloud/theia-trace-extension
[3] https://www.npmjs.com/package/traceviewer-base
[4] https://www.npmjs.com/package/traceviewer-react-components
[5] https://github.com/theia-ide/vscode-trace-extension/issues/82

**Comment by planger** [comment]
Thank you Bernd, I'm looking forward to this exciting contribution to the CDT Cloud ecosystem! Please let me know if you need help with anything along the contribution process.

**Comment by bhufmann** [comment]
I've started the contribution process by opening an issue at IPLab for IP checks:

https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/6928

**Comment by bhufmann** [comment]
Ticket to change GitHub organizations: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/2617
